### PR TITLE
CHeli and CRopes done

### PR DESCRIPTION
**Pull request by aap**

### CODE REVIEW

File: src/core/Ropes.cpp [hunk 1]
```diff
@@ -0,0 +1,169 @@
...
+}
+
+bool
+CRopes::CreateRopWithSwatComingDown(CVector pos)
+{
+	static uint32 ropeId = 0;
+
+	if(!CStreaming::HasModelLoaded(MI_SWAT) || !RegisterRope(ropeId+100, pos, true))
+		return false;
+	CCopPed *swat = (CCopPed*)CPopulation::AddPed(PEDTYPE_COP, 3, pos);	// TODO: 3 enum
```
Comment (erorcun) [inline src/core/Ropes.cpp:162]: COP_ARMY

File: src/peds/CopPed.h [hunk 2]
```diff
@@ -26,6 +26,8 @@ class CCopPed : public CPed
 	float m_fAbseilPos;
 	eCopType m_nCopType;
 	bool m_bThrowsSpikeTrap;
+	CEntity *m_pRopeEntity;	// CHeli or 1
+	uintptr m_nRopeID;
```
Comment (erorcun) [inline src/peds/CopPed.h:30]: why uintptr?

Comment (aap) [reply src/peds/CopPed.h:30]: id is calculated from CHeli pointer if there is one.

File: src/core/Ropes.cpp [hunk 3]
```diff
@@ -0,0 +1,169 @@
...
+			j = t;
+			f = t - j; 
+			*coors = (1.0f-f)*aRopes[i].m_pos[j] + f*aRopes[i].m_pos[j+1];
+			return true;
+		}
+	return false;
+}
+
+bool
+CRopes::CreateRopWithSwatComingDown(CVector pos)
```
Comment (erorcun) [inline src/core/Ropes.cpp:156]: CreateRop .d

File: src/core/Ropes.cpp [hunk 4]
```diff
@@ -0,0 +1,169 @@
...
+{
+	static uint32 ropeId = 0;
+
+	if(!CStreaming::HasModelLoaded(MI_SWAT) || !RegisterRope(ropeId+100, pos, true))
+		return false;
+	CCopPed *swat = (CCopPed*)CPopulation::AddPed(PEDTYPE_COP, COP_ARMY, pos);
```
Comment (erorcun) [inline src/core/Ropes.cpp:162]: oops I'm sorry, looks like 3 is a new value for swats abseiling, something like COP_HELI_SWAT. I can add it in my next PR if you want.

**Pull request merged**